### PR TITLE
remove `macos-12` runner label

### PR DESCRIPTION
**Pull request by shogo82148**
The macOS 12 runner image has been deprecated on 10/7/2024, and will be fully unsupported by 12/3/2024

- https://github.com/actions/runner-images/issues/10721
- https://github.blog/changelog/2024-08-19-notice-of-upcoming-deprecations-and-breaking-changes-in-github-actions-runners/

### CODE REVIEW

**Comment by rhysd** [comment]
Thanks. I believe we should merge this after 2024 12/3 and after this issue has been closed.

https://github.com/actions/runner-images/issues/10721

**Review by rhysd — CHANGES_REQUESTED** [review]
This review comment is to block the merge until the runner is completely dropped.

**Comment by AlexSkrypnyk** [comment]
Also, `macos-15` is available, but reported as false positive by actionlint.

**Comment by rhysd** [comment]
`macos-15` was supported at 1.7.3. If actionlint still complains about it, it should be a bug.

https://github.com/rhysd/actionlint/releases/tag/v1.7.3

**Pull request closed without merging**
